src/Nethermind/Nethermind.Serialization.Rlp/MyTxDecoder/BlobTxDecoder.cs

```diff
@@ -89,146 +61,74 @@ public void Decode(ref Transaction? transaction, int txSequenceStart, ReadOnlySp
-            }
+                DecodeShardBlobNetworkWrapper(transaction, ref decoderContext);
 
-            if ((rlpBehaviors & RlpBehaviors.ExcludeHashes) == 0)
-            {
-                transaction.Hash = CalculateHashForNetworkPayloadForm(transactionSequence);
-            }
-        }
-        else if ((rlpBehaviors & RlpBehaviors.ExcludeHashes) == 0)
-        {
-            if (transactionSequence.Length <= ITxDecoder.MaxDelayedHashTxnSize)
-            {
-                // Delay hash generation, as may be filtered as having too low gas etc
-                if (decoderContext.ShouldSliceMemory)
+                if ((rlpBehaviors & RlpBehaviors.AllowExtraBytes) == 0)
                 {
-                    // Do not copy the memory in this case.
-                    int currentPosition = decoderContext.Position;
-                    decoderContext.Position = txSequenceStart;
-                    transaction.SetPreHashMemoryNoLock(decoderContext.ReadMemory(transactionSequence.Length));
-                    decoderContext.Position = currentPosition;
+                    decoderContext.Check(networkWrapperCheck);
                 }
-                else
+
+                if ((rlpBehaviors & RlpBehaviors.ExcludeHashes) == 0)
                 {
-                    transaction.SetPreHashNoLock(transactionSequence);
+                    transaction.Hash = CalculateHashForNetworkPayloadForm(transactionSequence);
                 }
             }
-            else
+            else if ((rlpBehaviors & RlpBehaviors.ExcludeHashes) == 0)
             {
-                // Just calculate the Hash immediately as txn too large
-                transaction.Hash = Keccak.Compute(transactionSequence);
+                CalculateHash(transaction, txSequenceStart, transactionSequence, ref decoderContext);
             }
         }
     }
 
-    public void Encode(Transaction transaction, RlpStream stream, RlpBehaviors rlpBehaviors = RlpBehaviors.None, bool forSigning = false, bool isEip155Enabled = false, ulong chainId = 0)
+    protected override void EncodeTypedWrapped(Transaction transaction, RlpStream stream, RlpBehaviors rlpBehaviors, bool forSigning, int contentLength)
     {
-        int contentLength = GetContentLength(transaction, forSigning, rlpBehaviors.HasFlag(RlpBehaviors.InMempoolForm));
-        int sequenceLength = Rlp.LengthOfSequence(contentLength);
-
-        if ((rlpBehaviors & RlpBehaviors.SkipTypedWrapping) == 0)
-        {
-            stream.StartByteArray(sequenceLength + 1, false);
-        }
-
-        stream.WriteByte((byte)TxType.Blob);
-
         if (rlpBehaviors.HasFlag(RlpBehaviors.InMempoolForm))
         {
             stream.StartSequence(contentLength);
-            contentLength = GetContentLength(transaction, forSigning, false);
+            // we want internal content lenght here in normal form
+            contentLength = GetContentLength(transaction, rlpBehaviors & ~RlpBehaviors.InMempoolForm, forSigning);
```

Comment:
This block is a bit confusing, could you elaborate the comment a bit further?

Reply:
added more comments